viewer/packages/360-images/src/entity/Image360Entity.ts

```diff
@@ -56,14 +61,34 @@ export class Image360Entity implements Image360 {
     this._image360Icon = icon;
     this._image360VisualzationBox = new Image360VisualizationBox(this._transform, sceneHandler);
     this._image360VisualzationBox.visible = false;
+    this._requestRedraw = () => {};
   }
 
   /**
    * Loads the 360 image (6 faces) into the visualization object.
    */
   public async load360Image(abortSignal?: AbortSignal): Promise<void> {
-    const faces = await this._imageProvider.get360ImageFiles(this._image360Metadata.faceDescriptors, abortSignal);
-    await this._image360VisualzationBox.loadImages(faces);
+    const lowResolutionFaces = await this._imageProvider
+      .getLowResolution360ImageFiles(this._image360Metadata.faceDescriptors)
+      .catch(() => {
+        return undefined;
+      });
+
+    const fullResolutionFaces = this._imageProvider.get360ImageFiles(
+      this._image360Metadata.faceDescriptors,
+      abortSignal
+    );
+
+    if (!lowResolutionFaces) {
+      await this._image360VisualzationBox.loadImages(await fullResolutionFaces);
+    } else {
+      await this._image360VisualzationBox.loadImages(lowResolutionFaces);
+
+      fullResolutionFaces.then(async faces => {
+        await this._image360VisualzationBox.setFaceMaterials(faces);
+        this._requestRedraw();
```

Comment:
Could we await this promise outside this class instead of calling a callback when it finishes?